### PR TITLE
When attempting to derive component name from file name and file name…

### PR DESCRIPTION
**Pull request by gajus**
… is index.{ext} use parent directory name.

### CODE REVIEW

**Comment by gajus** [comment]
This is common when project structure is:

```
./src/components/{component name}/index.js
./src/components/{component name}/{... assets}
```

**Comment by gajus** [comment]
Is there a reason this PR is being stalled?

**Comment by gajus** [comment]
@sebmck Is there a reason this PR is being stalled?

File: src/index.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import path from 'path';
```
Comment (sebmck) [inline src/index.js]: Please adhere to the style in the file and use double quotes.

File: src/index.js [hunk 2]
```diff
@@ -43,11 +45,19 @@ export default function ({ Plugin, types: t }) {
 
     visitor: {
       ExportDefaultDeclaration(node, parent, scope, file) {
+        var displayName;
         if (isCreateClass(node.declaration)) {
-          addDisplayName(file.opts.basename, node.declaration);
+          displayName = file.opts.basename;
+
+          // ./{module name}/index.js
+          if (displayName === 'index') {
```
Comment (sebmck) [inline src/index.js]: Double quotes here too.

File: src/index.js [hunk 3]
```diff
@@ -43,11 +45,19 @@ export default function ({ Plugin, types: t }) {
 
     visitor: {
       ExportDefaultDeclaration(node, parent, scope, file) {
+        var displayName;
         if (isCreateClass(node.declaration)) {
-          addDisplayName(file.opts.basename, node.declaration);
+          displayName = file.opts.basename;
```
Comment (sebmck) [inline src/index.js]: Declare the variable here rather than at the top of the file.

**Comment by sebmck** [comment]
Sorry, this fell off my radar. Ping me when the changes commented inline are made and I'll push a new version. Thanks :blush:

File: src/index.js (hunk 3 above)
Comment (gajus) [inline src/index.js]: I don't understand this comment. I am assuming you are referring to `displayName` variable? You cannot do that in ES5 (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting).

Comment (gajus) [inline src/index.js]: It is unsafe, though considering that the rest of the code does it anyway, it makes little difference.

Comment (sebmck) [inline src/index.js]: You can totally do that. It's how JavaScript has worked since the first version.

**Comment by sebmck** [comment]
Squash your commits and this is good to go!

**Comment by gajus** [comment]
Done! :)

**Pull request merged**